File: LinkedLists/19 Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) 
    {
        int count=0;
        ListNode* p=head;
        ListNode* q=head;
        while(p!=NULL)
        {
            p=p->next;
            count+=1;
        }
        if(count==n)
        {
            head=head->next;
            return head;
        }
        for(int i=0;i<count-n-1;i++)
        {
            q=q->next;
        }
        q->next=q->next->next;
        return head;
    }
};

File: Trees/19 Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        p=head
        q=head
        c=0
        while(p!=None):
            p=p.next
            c+=1
        if(c==n):
            head=head.next
            return head
        for i in range(0,c-n-1):
            q=q.next
        q.next=q.next.next
        return head
